tests/test_conda_lock.py

```diff
@@ -2293,3 +2301,21 @@ def test_cli_version(capsys: "pytest.CaptureFixture[str]"):
     # the part before, in this case just "0.11.3".
     version_without_dev = __version__.split(".dev")[0]
     assert version_without_dev in result.stdout
+
+
+def test_pip_finds_recent_manylinux_wheels(
+    monkeypatch: "pytest.MonkeyPatch", lightgbm_environment: Path, conda_exe: str
+):
+    monkeypatch.chdir(lightgbm_environment.parent)
+    run_lock([lightgbm_environment], conda_exe=conda_exe, platforms=["linux-64"])
+    lockfile = parse_conda_lock_file(
+        lightgbm_environment.parent / DEFAULT_LOCKFILE_NAME
+    )
+
+    (lightgbm_dep,) = [p for p in lockfile.package if p.name == "lightgbm"]
+    manylinux_pattern = r"manylinux_(\d+)_(\d+).+\.whl"
+    manylinux_match = re.search(manylinux_pattern, lightgbm_dep.url)
+    assert manylinux_match, "No match found for manylinux version in {lightgbm_dep.url}"
+
+    manylinux_version = [int(each) for each in manylinux_match.groups()]
+    assert manylinux_version > [2, 17]
```

Comment:
Is this a "documentation"-type assertion or an actual test? In the latter case, what are we trying to check? (Like what would be the failure mode?)

Reply:
I am making sure that not we find a "recent" manylinux wheel i.e. more recent than manylinux_2_17 which was already covered by conda-lock before this PR.

Reply:
I tweaked the docstring and add a comment to try to make that more explicit

Reply:
Ah, ok, I think I get it now: before this PR the solver might have either:
1. chosen the wheel with the latest `lightgbm` version, which is what happened in #517
2. chosen an earlier version of `lightgbm` for which a `manylinux_2_17` wheel exists (not actually happening in any case, but theoretically possible)

Please correct me if I'm misunderstanding, and thanks for the clarification!